### PR TITLE
Bug 1162710-Compare chooser should offer a revision toolstips

### PR DESCRIPTION
**Pull request by MikeLing**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/749)

### CODE REVIEW

File: ui/partials/perf/comparechooserctrl.html [hunk 1]
```diff
@@ -21,11 +26,16 @@
...
           <label for="new-project-selector">Project</label>
-          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects"/>
+          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects" ng-change="updateNewRevisionTips()"/>
           <label for="new-revision-input">Revision</label>
             <div class="form-group" ng-class="{'has-error': newRevisionError}">
-            <input id="new-revision-input" maxlength="12" class="form-control" ng-model="newRevision">
-            <br/>
+            <input id="new-revision-input" maxlength="12" class="form-control" type="text" ng-model="newRevision"/>
+            <div class="input-group-btn">
+              <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Pushes<span class="caret"></span></button>
```
Comment (tojonmz) [inline ui/partials/perf/comparechooserctrl.html]: You might wish to list these maxlength, class, type, ng-model, etc attributes on new lines? So the line is not so long.

File: ui/partials/perf/comparechooserctrl.html [hunk 2]
```diff
@@ -21,11 +26,16 @@
...
+              <ul class="dropdown-menu dropdown-menu-right" >
+                <li ng-repeat="tip in newTipList"><a ng-click="getNewTipRevision(tip.revision)">{{tip.revision}} {{tip.author}}</a></li>
```
Comment (tojonmz) [inline ui/partials/perf/comparechooserctrl.html]: Similarly here, you might wish to next the anchor on a new line, inside the `li`.

I know a lot of the file is like this though.

File: ui/js/compareperf.js [hunk 3]
```diff
@@ -5,13 +5,46 @@
...
+        list.splice(0, list.length);
+        return $http.get (thServiceDomain + '/api/project/' +
+                          projectName + '/resultset/').then(function(resultSet) {
+                              resultSet['data'].results.forEach(function(revisionSet) {
+                                  var revisionTips = {};
+                                  revisionTips.revision = revisionSet.revision;
+                                  revisionTips.author = revisionSet.author;
+                                  list.push(revisionTips);
+                              });
+                       });
```
Comment (tojonmz) [inline ui/js/compareperf.js]: I think the indention is a bit out on this?

File: ui/js/compareperf.js [hunk 4]
```diff
@@ -5,13 +5,46 @@
...
+      }
+
+      $scope.updateOriRevisionTips = function () {
```
Comment (tojonmz) [inline ui/js/compareperf.js]: I'd check with Will, perhaps we'd want "Orig" here, rather than "Ori".

**Comment by tojonmz** [comment]
I haven't tested the branch, but I've left a few comments on the code fwiw :)

File: ui/js/compareperf.js [hunk 5]
```diff
@@ -5,13 +5,46 @@
...
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
       $scope.originalProject = $scope.newProject = projects[0];
+      $scope.oriTipList = [];
+      $scope.newTipList = [];
+      var getRevisionTips = function (projectName, list) {
+        // due to we push the revision data into list,
+        // so we need clear the data before we push new data into it.
+        list.splice(0, list.length);
+        return $http.get (thServiceDomain + '/api/project/' +
```
Comment (wlach) [inline ui/js/compareperf.js]: You should use ThResultSetModel here. Something like:

    ThResultSetModel.getResultSets(projectName).then(function(response) {
        var resultsets = response.data.results;
        ...

File: ui/js/compareperf.js [hunk 6]
```diff
@@ -5,13 +5,46 @@
...
 perf.controller('CompareChooserCtrl', [
-  '$state', '$stateParams', '$scope', 'ThRepositoryModel', 'ThResultSetModel',
-  function CompareChooserCtrl($state, $stateParams, $scope,
+  '$state', '$stateParams', '$scope', '$http', 'ThRepositoryModel', 'ThResultSetModel',
+  function CompareChooserCtrl($state, $stateParams, $scope, $http,
                               ThRepositoryModel, ThResultSetModel) {
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
       $scope.originalProject = $scope.newProject = projects[0];
+      $scope.oriTipList = [];
```
Comment (wlach) [inline ui/js/compareperf.js]: ori -> original, everywhere in the file. It's confusing to use abbreviations in one place and not others.

File: ui/js/compareperf.js (hunk 5 above)
Comment (wlach) [inline ui/js/compareperf.js]: Also, there should be no space between a function call (or declaration) and the parentheses:

i.e. always do:

    function foo(bar) { ...
    }
    foo(123);

and not:

    function foo (bar) { ... // WRONG
    }
    foo (123); // WRONG

File: ui/js/compareperf.js [hunk 7]
```diff
@@ -5,13 +5,46 @@
...
+        // due to we push the revision data into list,
+        // so we need clear the data before we push new data into it.
+        list.splice(0, list.length);
+        return $http.get (thServiceDomain + '/api/project/' +
+                          projectName + '/resultset/').then(function(resultSet) {
+                              resultSet['data'].results.forEach(function(revisionSet) {
+                                  var revisionTips = {};
+                                  revisionTips.revision = revisionSet.revision;
+                                  revisionTips.author = revisionSet.author;
+                                  list.push(revisionTips);
```
Comment (wlach) [inline ui/js/compareperf.js]: Write this as:

    list.push({
        revision: revisionSet.revision,
        author: revisionSet.author
    })

File: ui/partials/perf/comparechooserctrl.html [hunk 8]
```diff
@@ -21,11 +27,19 @@
...
-          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects"/>
+          <select id="new-project-selector" class="form-control" ng-model="newProject" ng-options="project.name for project in projects" ng-change="updateNewRevisionTips()"/>
           <label for="new-revision-input">Revision</label>
             <div class="form-group" ng-class="{'has-error': newRevisionError}">
-            <input id="new-revision-input" maxlength="12" class="form-control" ng-model="newRevision">
-            <br/>
+            <input id="new-revision-input" maxlength="12" class="form-control" type="text" ng-model="newRevision"/>
+            <div class="input-group-btn">
+              <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+              Pushes<span class="caret"></span></button>
```
Comment (wlach) [inline ui/partials/perf/comparechooserctrl.html]: I think I prefer the term `Recent` to `Pushes`, as it's more precise.

Also, there should be a space between the text and the caret. I.e.

    Pushes <span class="caret"/>

File: ui/js/compareperf.js [hunk 9]
```diff
@@ -5,13 +5,46 @@
...
+  '$state', '$stateParams', '$scope', '$http', 'ThRepositoryModel', 'ThResultSetModel',
+  function CompareChooserCtrl($state, $stateParams, $scope, $http,
                               ThRepositoryModel, ThResultSetModel) {
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
       $scope.originalProject = $scope.newProject = projects[0];
+      $scope.originalTipList = [];
+      $scope.newTipList = [];
+      var getRevisionTips = function(projectName, list) {
+        // due to we push the revision data into list,
```
Comment (jmaher) [inline ui/js/compareperf.js]: nit: in this added code we are mixing 2 and 4 space indentation.  line 17 is 2 space indentation, then on line 21 we do 4 space indentation.

**Pull request closed without merging**